apps/application-system/api/src/app/modules/application/files/file.service.ts

```diff
@@ -0,0 +1,126 @@
+import { Injectable } from '@nestjs/common'
+import { generateResidenceChangePdf } from './utils/pdf'
+import * as AWS from 'aws-sdk'
+import { uuid } from 'uuidv4'
+import { PDF_TYPES } from '@island.is/application/api-template-utils'
+import { Application } from './../application.model'
+import { FormValue } from '@island.is/application/core'
+import {
+  ParentResidenceChange,
+  PersonResidenceChange,
+} from '@island.is/application/templates/children-residence-change'
+import { User } from '@island.is/api/domains/national-registry'
+import { environment } from '../../../../environments'
+
+@Injectable()
+export class FileService {
+  s3: AWS.S3
+  private one_minute = 60
+
+  constructor() {
+    this.s3 = new AWS.S3()
+  }
+
+  async createPdf(application: Application, type: PDF_TYPES): Promise<string> {
+    const answers = application.answers as FormValue
+    const externalData = application.externalData as FormValue
+
+    switch (type) {
+      case PDF_TYPES.CHILDREN_RESIDENCE_CHANGE: {
+        const {
+          parentA,
+          parentB,
+          childrenAppliedFor,
+          expiry,
+        } = this.variablesForResidenceChange(answers, externalData)
+        return await this.createResidenceChangePdf(
+          parentA,
+          parentB,
+          childrenAppliedFor,
+          expiry,
+        )
+      }
+    }
+  }
+
+  variablesForResidenceChange(answers: FormValue, externalData: FormValue) {
+    const parentBNationalRegistry = externalData.parentNationalRegistry as FormValue
+    const nationalRegistry = externalData.nationalRegistry as FormValue
+    const nationalRegistryData = (nationalRegistry.data as unknown) as User
+    const childrenAppliedFor = (answers.selectChild as unknown) as Array<
+      PersonResidenceChange
+    >
+    const parentB = (parentBNationalRegistry.data as unknown) as ParentResidenceChange
+
+    parentB.email = answers.parentBEmail as string
+    parentB.phoneNumber = answers.parentBPhoneNumber as string
+
+    const parentA: ParentResidenceChange = {
+      id: nationalRegistryData.nationalId,
+      name: nationalRegistryData.fullName,
+      ssn: nationalRegistryData.nationalId,
+      phoneNumber: answers.phoneNumber as string,
+      email: answers.email as string,
+      address: nationalRegistryData.address?.streetAddress as string,
+      postalCode: nationalRegistryData.address?.postalCode as string,
+      city: nationalRegistryData.address?.city as string,
+    }
```

Comment:
I added tests for this. But is there a way to get type safety for this? Or do we always have to this thingamajig and trust the type will be cast to the correct type?